### PR TITLE
Beef up OrderedTree API

### PR DESCRIPTION
**Pull request by privat**
This add services to the class OrderedTree

### CODE REVIEW

File: lib/ordered_tree.nit [hunk 1]
```diff
@@ -69,20 +70,67 @@ class OrderedTree[E: Object]
 	# For each element, the ordered array of its direct sub-elements.
 	var sub = new HashMap[E, Array[E]]
 
+	# For each element, the parent
```
Comment (Morriar) [inline lib/ordered_tree.nit]: This sentence is as weird as your language :)

File: lib/ordered_tree.nit [hunk 2]
```diff
@@ -69,20 +70,67 @@ class OrderedTree[E: Object]
...
+	# assert tree.parent(2) == 1
+	# assert tree.parent(1) == null
+	# ~~~
+	fun parent(e: E): nullable E do return parents[e]
+
 	# Add a new element `e` in the tree.
 	# `p` is the parent of `e`.
 	# if `p` is null, then `e` is a root element.
+	#
+	# If `e` is already in the tree, it is removed before
```
Comment (Morriar) [inline lib/ordered_tree.nit]: Punctuation

File: lib/ordered_tree.nit [hunk 3]
```diff
@@ -69,20 +70,67 @@ class OrderedTree[E: Object]
...
 	end
 
+	# Append all nodes `es` as children of `p`.
+	fun add_all(p: nullable E, es: Collection[E])
+	do
+		for e in es do add(p, e)
+	end
+
+	# Temporary remove `e`
+	# Children of `e` a let as is thus make the tree inconstant until `e` is added back.
```
Comment (Morriar) [inline lib/ordered_tree.nit]: Harlette?

Comment (Morriar) [inline lib/ordered_tree.nit]: are let*

File: lib/ordered_tree.nit [hunk 4]
```diff
@@ -69,20 +70,67 @@ class OrderedTree[E: Object]
...
+	do
+		var old_parent = parents.get_or_null(e)
+		if old_parent != null then
+			sub[old_parent].remove(e)
+		else if roots.has(e) then
+			roots.remove(e)
+		end
+	end
+
+	redef fun has(e)
```
Comment (Morriar) [inline lib/ordered_tree.nit]: oneliner

File: lib/ordered_tree.nit [hunk 5]
```diff
@@ -165,21 +208,50 @@ class OrderedTree[E: Object]
...
+	#     tree.add_all(12, [121, 122])
+	#     tree.add_all(2, [21, 22])
 	#     var order = [1, 11, 111, 112, 12, 121, 122, 2, 21, 22]
-	#     var res = new Array[Int]
-	#     for i in tree do res.add(i)
-	#     assert res == order
+	#     assert tree.iterator.to_a == order
 	redef fun iterator do return new OrderedTreeIterator[E](self)
+
+	# Two trees are equals if they have the same nodes in the same order
```
Comment (Morriar) [inline lib/ordered_tree.nit]: equal*

**Comment by privat** [comment]
reroll: spelling by @Morriar , Also fix clone that was buggy (added a nitunit)

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**